exercises/resistor-color-trio/.meta/src/reference/java/ResistorColorTrio.java

```diff
@@ -0,0 +1,17 @@
+class ResistorColorTrio {
+    private enum Color {
+        black, brown, red, orange, yellow, green, blue, violet, grey, white;
+    };
+
+    String value(String[] colors) {
+        int intValue = (10 * Color.valueOf(colors[0]).ordinal() + Color.valueOf(colors[1]).ordinal())
+                * (int) Math.pow(10, Color.valueOf(colors[2]).ordinal());
+        String stringValue;
+        if (intValue >= 1000) {
+            stringValue = String.valueOf(intValue / 1000) + " kiloohms";
+        } else {
+            stringValue = String.valueOf(intValue) + " ohms";
+        }
+        return stringValue;
```

Comment:
The canonical implementation should probably include encoding more information into the Color enum. Might also want to consider another enum for the powers (like OHMS, KILOOHMS, ...). This solution only adds support for kiloohms and we can do better (I know the Kotlin version goes up to something like exaohms).

Reply:
The [Haskell solution](https://github.com/exercism/haskell/pull/869) only goes to giga, but adds [optional edge cases](https://github.com/exercism/haskell/pull/869/files#diff-6f477c701557f26d2c698dbce05157abR88) that the original exercise does not address, because functional programmers have an obsession with [totality](https://en.wikipedia.org/wiki/Partial_function#Total_function). The point is that while you could extend the range beyond giga, you could also do a lot of other things, or not.

Reply:
I suppose I am biased by having seen the Kotlin implementation of this exercise. IMO: we should go up to giga specifically since the exercise naturally produces gigaohms.

Reply:
Keeping it simple and as specific as possible seemed like a good option.
However, as per my other comment, i do not know if this is preferrable or not.